fxgl-trade/src/main/kotlin/com/almasb/fxgl/inventory/Inventory.kt

```diff
@@ -28,32 +29,65 @@ class Inventory<T>(var capacity: Int) {
     private val items = FXCollections.observableArrayList<T>()
     private val itemsProperty = FXCollections.unmodifiableObservableList(items)
 
-    private val itemsData = hashMapOf<T, ItemData<T>>()
+    private val itemsData = hashMapOf<T, ObservableList<ItemData<T>>>()
+
+    private var stackMax = Int.MAX_VALUE
 
     /**
      * Number of items in inventory.
      */
     val size: Int
-        get() = items.size
+        get() {
+            var totalSize = 0
```

Comment:
I could make that change but as the PR is, it would let the user add an arbitrary amount of stacks. `capacity` would be only a bound for number of item types, `T`, in the inventory. It would also make code like this in `incrementQuantity` unnecessary:

https://github.com/AlmasB/FXGL/blob/1c45b84d837322b9db72e4a68982ef544b7b06cd/fxgl-trade/src/main/kotlin/com/almasb/fxgl/inventory/Inventory.kt#L184-L194

Reply:
I see, in which case we will need to clearly define what `size` and other parameters are in the context of Inventory. Consider this comment resolved.